lib/env.c

```diff
@@ -52,31 +52,49 @@ static char * const noslash[] = {
 
 
 struct ul_env_list {
-	char *env;
+	char	*name;
+	char	*value;
+
 	struct ul_env_list *next;
 };
 
+
 /*
- * Saves @name env.variable to @ls, returns pointer to the new head of the list.
+ * Saves env.variable to @ls, returns pointer to the new head of the list.
+ *
+ * @str is "name" (in this case @val has to be set) or "name=value"
  */
-static struct ul_env_list *env_list_add(struct ul_env_list *ls0, const char *str)
+static struct ul_env_list *env_list_add(struct ul_env_list *ls0,
+					const char *str,
+					const char *val)
```

Comment:
Instead of overloading for `val == NULL` I'd prefer a new specific function for that.

Reply:
The API will be pretty hairy. I'll think about it.

Reply:
I have split the function(s).